capirca/lib/fortigate.py

```diff
@@ -0,0 +1,330 @@
+    for group in addresses:
+      for addr in group:
+        if addr and not isinstance(addr, nacaddr.IPv6):
+          self._add_address_to_fw_addresses(addr.with_prefixlen)
+
+  def __str__(self):
+    lines = []
+
+    self._generate_address_names(self._term.destination_address, self._term.source_address)
+    # lines.extend(self.firewall_addresses)
+
+    dest_addresses = self._get_addresses_name(self._term.destination_address)
+    src_addresses = self._get_addresses_name(self._term.source_address)
+    services = self._get_services_string(self._term.protocol, self._term.destination_port)
+
+    lines.append('\t\tset comments %s' % self._term.name)
+    lines.append('\t\tset srcintf %s' % (self._term.source_interface or 'any'))
+    lines.append('\t\tset dstintf %s' % (self._term.destination_interface or 'any'))
+    lines.append('\t\tset dstaddr %s' % dest_addresses)
+    lines.append('\t\tset srcaddr %s' % src_addresses)
+    lines.append('\t\tset action %s' % _ACTION_TABLE.get(self._term.action[0]))
+    lines.append('\t\tset service %s' % services)
+    lines.append('\t\tset schedule always')
+    if self._term.logging:
+      lines.append('\t\tset logtraffic all')
+
+    return '\n'.join(lines)
+
+
+class Fortigate(aclgenerator.ACLGenerator):
+  """A cisco policy object."""
+
+  _PLATFORM = 'fortigate'
+  _DEFAULT_PROTOCOL = 'ALL'
+  SUFFIX = '.fcl'
+  # Protocols should be emitted as numbers.
+  _PROTO_INT = True
+  _TERM_REMARK = True
+
+  def _BuildTokens(self):
+    """Build supported tokens for platform.
+
+    Returns:
+      tuple containing both supported tokens and sub tokens
+    """
+    supported_tokens, supported_sub_tokens = super(Fortigate, self)._BuildTokens()
+
+    supported_tokens |= {'source_interface',
+                         'destination_interface',
+                         'logging'}
+
+    supported_sub_tokens.update({'option': {'from_id'},
+                                 # Warning, some of these are mapped
+                                 # differently. See _ACTION_TABLE
+                                 'action': {'accept', 'deny', 'reject',
+                                            'reject-with-tcp-rst'}})
+    return supported_tokens, supported_sub_tokens
+
+  def _TranslatePolicy(self, pol, exp_info):
+    self.fortigate_policies = []
+    current_date = datetime.datetime.utcnow().date()
+    exp_info_date = current_date + datetime.timedelta(weeks=exp_info)
+
+    # a mixed filter outputs both ipv4 and ipv6 acls in the same output file
+
+    for header, terms in pol.filters:
+      if self._PLATFORM not in header.platforms:
+        continue
+
+      filter_options = header.FilterOptions(self._PLATFORM)
+
+      if (len(filter_options) < 2 or filter_options[0] != "from-id"):
+        raise UnsupportedFilterError(
+          "Fortigate Firewall filter arguments must specify from_id")
+
+      from_id = filter_options[1]
+      Term.CURRENT_ID = int(from_id)
+
+      self.verbose = True
```

Comment:
Do you actually use verbose in here anywhere? This is used to toggle comments in the generated syntax.